nemo/collections/asr/parts/utils/online_clustering.py

```diff
@@ -1189,3 +1196,422 @@ def forward_infer(
+        else:
+            emb_part = emb[self.unit_window_len * index : self.unit_window_len * (index + 1)]
+            offset_index = self.unit_window_len * index
+        return emb_part, offset_index
+
+    def forward_unit_infer(
+        self,
+        emb: torch.Tensor,
+        oracle_num_speakers: int = -1,
+        max_rp_threshold: float = 0.15,
+        max_num_speakers: int = 8,
+        sparse_search_volume: int = 30,
+        min_samples_for_nmesc: int = 1,
+        fixed_thres: float = -1.0,
+        cuda: bool = False,
+    ) -> torch.LongTensor:
+        """
+        This function is slimmed down version of forward_infer function in `SpeakerClustering` class.
+        Please refer to `SpeakerClustering` class for the original argument information.
+        """
+        if emb.shape[0] == 1:
+            return torch.zeros((1,), dtype=torch.int64)
+
+        mat = getCosAffinityMatrix(emb)
+        nmesc = NMESC(
+            mat,
+            max_num_speakers=max_num_speakers,
+            max_rp_threshold=max_rp_threshold,
+            sparse_search=self.sparse_search,
+            sparse_search_volume=sparse_search_volume,
+            fixed_thres=fixed_thres,
+            nme_mat_size=self.nme_mat_size,
+            maj_vote_spk_count=self.maj_vote_spk_count,
+            parallelism=self.parallelism,
+            cuda=cuda,
+            device=self.device,
+        )
+        if mat.shape[0] > min_samples_for_nmesc:
+            est_num_of_spk, p_hat_value = nmesc.forward()
+            affinity_mat = getAffinityGraphMat(mat, p_hat_value)
+        else:
+            est_num_of_spk = torch.tensor(1)
+            affinity_mat = mat
+
+        if oracle_num_speakers > 0:
+            n_clusters = int(oracle_num_speakers)
+        else:
+            n_clusters = int(est_num_of_spk.item())
+
+        spectral_model = SpectralClustering(n_clusters=n_clusters, cuda=self.cuda, device=self.device)
+        Y = spectral_model.forward(affinity_mat)
+        return Y
+
+    def forward(self, param_dict: Dict[str, torch.Tensor]) -> torch.LongTensor:
+        """
+        A function wrapper designed for performing inference using an exported script format.
+
+        Note:
+            A dictionary is used to facilitate inference with the exported jit model in the Triton server. 
+            This is done using an easy-to-understand naming convention.
+            See https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_configuration.md#special-conventions-for-pytorch-backend
+
+        Args:
+            param_dict (dict):
+                Dictionary containing the arguments for speaker clustering.
+                See `forward_infer` function for the argument information.
+
+        Returns:
+            Y (LongTensor):
+                Speaker labels for the segments in the given input embeddings.
+        """
+        embeddings_in_scales = param_dict['embeddings']
```

Comment:
Will this function be directly called by the user? If so, do we need to add some value checking statements to for `param_dict` such that proper error messages will be thrown if invalid or missing values are found?

Reply:
Added dimension check in `split_input_data` function in `offline_clustering.py`.